### PR TITLE
Ensure the connection is tried several times

### LINKED ISSUE #80: Make sure different routes are tried
While investigating the failure in https://github.com/c4dt/lightarti-rest/runs/3963994558?check_suite_focus=true#step:9:85, @laurent-girod wrote:

```
This is the incriminated relay, apparently, its maintainer changed its ORPort from 443 to 993.
[...]
So, if you wait for the next consensus, the problem might solve itself.
(or by rerunning immediately, as Arti randomly selects the routers to build a circuit.)
```

What is strange is that I thought it'll try three circuits anyway. The test is here:
https://github.com/c4dt/lightarti-rest/blob/ca06e663fdae03b17f53bb4852275f45577eb994/src/lightarti.rs#L95
and the retry is here:
https://github.com/c4dt/lightarti-rest/blob/ca06e663fdae03b17f53bb4852275f45577eb994/src/lightarti.rs#L53
so why doesn't it choose a new circuit?

### PR DESCRIPTION
**Pull request by cgrigis**
Ensure the connection is tried several times no matter where the error occurs.

Fixes #80

### CODE REVIEW

File: src/lightarti.rs [hunk 1]
```diff
@@ -45,44 +47,60 @@ pub fn tls_send(host: &str, request: &str, cache: &Path) -> Result<String> {
...
         .context("bootstrap tor client")?;
 
-        send_request(tor_client, host, request).await
+        send_request(&tor_client, host, request).await
     })
 }
 
-/// Sends a GET request over a TLS connection and returns the result.
-async fn send_request(tor: TorClient<impl Runtime>, host: &str, request: &str) -> Result<String> {
+/// Tries several times to send a request, after which returns an error if still unsuccessful.
```
Comment (ineiti) [inline src/lightarti.rs:54]:
```suggestion
/// Tries several times to send a request, after which it returns an error if still unsuccessful.
```

**Review by ineiti — APPROVED** [review]
LGTM - I see there are some `prettyfications` - don't we have a `prettier` in the CI/CD? If not, I'll open a new issue for that.

**Comment by cgrigis** [comment]
> LGTM - I see there are some `prettyfications` - don't we have a `prettier` in the CI/CD? If not, I'll open a new issue for that.

I guess not...

File: src/lightarti.rs (hunk 1 above)
Comment (cgrigis) [reply src/lightarti.rs]: Changed the comment a bit

**Pull request merged**